libraries/install_plugin.rb

```diff
@@ -0,0 +1,62 @@
+module Extensions
+
+  # Install an Elasticsearch plugin
+  #
+  # In the simplest form, just pass a plugin name in the GitHub <user>/<repo> format:
+  #
+  #     install_plugin 'karmi/elasticsearch-paramedic'
+  #
+  # You may also optionally pass a version:
+  #
+  #     install_plugin 'elasticsearch/elasticsearch-mapper-attachments', 'version' => '1.6.0'
+  #
+  # ... as well as the URL:
+  #
+  #     install_plugin 'hunspell', 'url' => 'https://github.com/downloads/.../elasticsearch-analysis-hunspell-1.1.1.zip'
+  #
+  # The "elasticsearch::plugins" recipe will install all plugins listed in
+  # the role/node attributes or in the data bag (`node.elasticsearch.plugins`).
+  #
+  # Example:
+  #
+  #     { elasticsearch: {
+  #         plugins: {
+  #           'karmi/elasticsearch-paramedic' => {},
+  #           'lukas-vlcek/bigdesk'           => { 'version' => '1.0.0' },
+  #           'hunspell'                      => { 'url' => 'https://github.com/downloads/...' }
+  #         }
+  #       }
+  #     }
+  #
+  # See <http://wiki.opscode.com/display/chef/Setting+Attributes+(Examples)> for more info.
+  #
+  def install_plugin name, params={}
+
+    ruby_block "Install plugin: #{name}" do
+      block do
+        version = params['version'] ? "/#{params['version']}" : nil
+        url     = params['url']     ? " -u #{params['url']}" : nil
+
+        command = "#{node.elasticsearch[:env][:es_home]}/bin/plugin -i #{name}#{version}#{url}"
+        Chef::Log.debug command
+
+        raise "[!] Failed to install plugin" unless system command
```

Comment:
I think is possible, but i use the same code of https://github.com/elastic/cookbook-elasticsearch/blob/master/libraries/install_plugin.rb and it works and i don't understand why change this.